src/main/java/parser/Parser.java

```diff
@@ -0,0 +1,114 @@
+package parser;
+
+import data.exception.DukeException;
+
+public class Parser {
+    /**
+     * Commands to be run based on the user's input.
+     */
+    public enum Command {
+        BYE,
+        LIST,
+        MARK,
+        UNMARK,
+        DELETE,
+        FIND,
+        TODO,
+        DEADLINE,
+        EVENT
+    }
+
+    /**
+     * Returns a command to be executed given a specified input provided by the user.
+     *
+     * @param input The input of the user.
+     * @return A command to be exectued which is based on the user input.
+     * @throws DukeException If the first word of the input is not a valid command.
+     */
+    public Command parse(String input) throws DukeException {
+        String[] inputSplit = input.split(" ");
+        Command command = this.getCommand(input);
+        switch(command) {
+        case BYE:
+            return Command.BYE;
+        case LIST:
+            return Command.LIST;
+        case MARK:
+            if (inputSplit.length == 1) {
+                throw new DukeException("Input the task number");
+            } else if (inputSplit.length > 2) {
+                throw new DukeException("Invalid command");
+            }  else {
+                return Command.MARK;
+            }
+        case UNMARK:
+            if (inputSplit.length == 1) {
+                throw new DukeException("Input the task number");
```

Comment:
Perhaps you can consider using a named constant for `"Input the task number"` and `"Invalid command"`?

Reply:
Yep ok makes sense